src/main/java/com/vmware/vipclient/i18n/base/cache/LocaleCacheItem.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n.base.cache;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class LocaleCacheItem extends FormatCacheItem implements CacheItem {
+
+    private final Map<String, String> cachedData = new HashMap<String, String>();
+
+    public LocaleCacheItem() {
+
+    }
+
+    public LocaleCacheItem (Map<String, String> dataMap, String etag, long timestamp, Long maxAgeMillis) {
+        this.set(dataMap, etag, timestamp, maxAgeMillis);
+    }
+
+    public synchronized void set(Map<String, String> dataMap, long timestamp) {
+        this.set(dataMap, null, timestamp, null);
+    }
+
+    public synchronized void set(Map<String, String> dataToCache, String etag, long timestamp, Long maxAgeMillis) {
+        if (dataToCache != null)
+            this.cachedData.putAll(dataToCache);
+        this.set(etag, timestamp, maxAgeMillis);
+    }
+
+    public synchronized void set(String etag, long timestamp, Long maxAgeMillis) {
+        if (etag != null && !etag.isEmpty())
+            setEtag(etag);
+        setTimestamp(timestamp);
+        if (maxAgeMillis != null)
+            setMaxAgeMillis(maxAgeMillis);
+    }
+
+    public synchronized void set(LocaleCacheItem cacheItem) {
+        this.set(cacheItem.getCachedData(), cacheItem.getEtag(), cacheItem.getTimestamp(), cacheItem.getMaxAgeMillis());
+    }
```

Comment:
I remove this method as it's not used. 
But I wonder if the CacheItem properties can be modified individually? Because no individually setter methods are provided.

Reply:
Hmm, I think I do not understand your comment. The point is to NOT allow individual modification of these properties, so 1 thread will lock the object first, set all three, then release the lock.